proposals/new-ipfs-gc.md

```diff
@@ -0,0 +1,103 @@
+- Need to be able to remove specific content 
+
+This project is critical for continued growth of users such as Pinata, as the current GC implementation is a significant obstacle for them.  The project is also important to users who need to remove specific content from their nodes, but do not want complete GC.
+
+### Assumptions and Hypotheses
+
+For this project to matter, it must significantly improve operations for highly impacted customers like Pinata. Faster GC will have a directly improve Pinata’s use of IPFS.  Having the ability to remove content when it becomes unpinned will further enhance the improvement for Pinata’s use case, as GC can avoid looking at all blocks and can operate on only those that became unpinned. 
+For users who do not see or care about the GC performance issues, this project will still matter to users who need to remove specific content from nodes, but do not want to remove all unpinned content.
+
+### User workflow example
+
+#### Bulk GC
+A user would use the new GC in the same way as the previous version, but with potentially significant reductions in duration and therefore less disruption to IPFS operation.  This may eliminate the need to work around the disruption by doing things like taking nodes offline to perform GC.  Ideally, GC lock time should be minimized to allow GC to be done effectively concurrently with normal IPFS operation.
+
+#### Removing Specific Content
+Certain users may only need to remove content that becomes unpinned.  This is particularly true of pinning services such as Pinata, and will be their primary, if not only, method of garbage collection. To remove content that becomes unpinned, the command line workflow is:
+- Unpin the desired content: `ipfs pin rm <cid>`
+- GC the unpinned content: `ipfs repo gc --cid=<cid>`
+
+The above could also be done with a single command, if it adds value: `ipfs pin rm -gc <cid>`
+It may be valuable once local named pinning is available.
+
+In the case that a user wants to remove content that was never pinned, they execute step 2 from above. This tells GC to walk the IPLD DAG identified by the CID and remove any associated blocks that are not referenced.
+
+### Impact
+While impact varies greatly for different types of users, some of our most important users (e.g. Pinata) are heavily impacted and have a need for this feature.  It will have a significant effect on their business operations.  Therefore, this gets an impact rating of 3.
+🔥🔥🔥 
+
+### Leverage
+This fixes a few specific issues, but is not highly leverageable.  The ability to remove specific content may allow implementation of a custom data retention policy.  Therefore this gets a 1 for leverage. 
+🎯
+
+### Confidence
+Medium (level 5) confidence, according to [this scale](https://medium.com/@nimay/inside-product-introduction-to-feature-priority-using-ice-impact-confidence-ease-and-gist-5180434e5b15)_.
+
+## Project definition
+
+### Brief plan of attack
```

Comment:
Can you also link to prior work here? Afaik we’ve already done some experiments

Reply:
This issue tracks relates issues: https://github.com/ipfs/go-ipfs/issues/7752

Reply:
Added a github link that tracks related issues and documents.  I have also done some prototyping on a different implementation approach, but that did not pan out,  for reasons that this new design specifically tries to avoid.  I did not include that doc, as I think it would end up being more confusing and not adding anything beyond what is already in the new design doc.

Reply:
thanks!